draft-ietf-quic-transport.md

```diff
@@ -3874,145 +3874,136 @@ The details of loss detection and congestion control are described in
-application data packet number space will be increased by two.
-
+ECN counts.  Coalesced QUIC packets (see {{packet-coalesce}}) share the same IP
+header.  The ECN counts for the ECN codepoint received in the associated IP
+header are incremented once for each QUIC packet.
 
-### ECN Validation {#ecn-validation}
+For example, if one each of an Initial, 0-RTT, Handshake, and 1-RTT QUIC packet
+are coalesced, the corresponding counts for the Initial and Handshake packet
+number spaces will be incremented by one each and the counts for the
+application data packet number space will be increased by two.
 
-It is possible for faulty network devices to corrupt or erroneously drop packets
-with ECN markings.  To provide robust connectivity in the presence of such
-devices, each endpoint independently validates ECN counts and disables ECN if
-the path is not showing consistent support for ECN.
+An ECN count is not incremented when no QUIC packets from the received IP
+packet are processed. A QUIC packet detected by a receiver as a duplicate also
+does not increase an ECN count; see {{security-ecn}} for relevant security
+concerns.
 
-Endpoints validate ECN for packets sent on each network path independently.  An
-endpoint thus validates ECN on new connection establishment, when switching to a
-server's preferred address, and on active connection migration to a new path.
-{{ecn-alg}} describes one possible algorithm for testing paths for ECN support.
 
-Even if an endpoint does not use ECN markings on packets it transmits, the
-endpoint MUST provide feedback about ECN markings received from the peer if they
-are accessible.  Failing to report ECN counts will cause the peer to disable ECN
-marking.
+### ECN Validation {#ecn-validation}
 
+It is possible for faulty network devices to corrupt or erroneously drop
+packets that carry a non-zero ECN codepoint. To ensure connectivity in the
+presence of such devices, an endpoint validates the ECN counts for each network
+path and disables use of ECN on that path if errors are detected.
 
-#### Sending ECN Markings
+To perform ECN validation for a new path:
 
-To start ECN validation, an endpoint SHOULD do the following when sending
-packets on a new path to a peer:
+* The endpoint sets an ECT(0) codepoint in the IP header of early outgoing
+  packets sent on a new path to the peer ({{!RFC8311}}).
 
-* Set the ECT(0) codepoint in the IP header of early outgoing packets sent on a
-  new path to the peer ({{!RFC8311}}).
+* The endpoint monitors whether all packets sent with an ECT codepoint are
+  eventually deemed lost (Section 6 of {{QUIC-RECOVERY}}), indicating
+  that ECN validation has failed.
 
-* If all packets that were sent with the ECT(0) codepoint are eventually deemed
-  lost (Section 6 of {{QUIC-RECOVERY}}), validation is deemed to have failed.
+If an endpoint has cause to expect that IP packets with an ECT codepoint might
```

Comment:
This isn't a departure from existing text -- see lines 3953-3956 in the old text.

Reply:
So, I’m going to chime-in here. Indeed it was in the “original” text, which is why I included this. However, we now have several people (me included) who doubt this is yet correct, so here are my thoughts:

(1)  I think it *is* important to be robust and describe how the protocol falls back in the corner case where the path drops IP packets with a non-zero ECN field - and we should very clearly say this.
(2) I think this can actually happen, but we should *not* make it sound like this often happens in general. I’m confident this is not “normal”.
(3) If we need to *measure* the path characteristic, using 10 packets seems a good test to me. However, as a fall back detection mechanism in a transport protocol this seems odd to stop using ECN after 10 packets when we expect that ECT is forwarded. Note: The text says {{ecn-ack}} says 10, the current text does not.

So.... This is a proposed change: We seem to be converging on clear text, would it be possible to remove the "10 packets" and revise this mechanism to say that the ECN counts need to increment, and pathological loss with no incrementing of counts is an indication that the sender should become “failed” for this connection? I think the text already virtually says this.

Reply:
"Pathological loss" isn't going to be good enough as a mechanism, and there's a rathole waiting there to define precisely what "pathological" means. We chose 10 packets based on the IW, and also because we wanted to start somewhere and then change the mechanism as we learned more through experience.

I suggest that we don't muddle this particular (large) PR with an additional, potentially-contentious discussion about a mechanism change.

Reply:
I do like Jana's proposal to not muddle this particular (large) PR with this "issue". Would others agree to finalise the PR and then discuss this as a separate topic - afterwards?

Reply:
I think we discussed this all ready once. I'm fine with 10 packet as it's the initial window (but maybe that's what we should say), however, given the result is the same I would rather state it as enable ECT marking and if all packet of your first flight are lost validation fails and you disable. Describing this specifically as probably makes it just sounds more complicated that it is.

Reply:
The github diff didn't line up well, hence my confusion.  The way this is described still seems a bit unrealistic to me, but I can see this PR doesn't substantively change it, so I'm fine with that.